### PR TITLE
Problem: error message is not explicit when allow-picked-version = false

### LINKED ISSUE #481: Improve error message when allow-picked-version = false
When you set "allow-picked-versions=false" you get a message like:
```
Error: Picked: pytest-instafail = 0.4.1
```
... which is not helping much.

Luckily, I found help in the plone docs, cf https://docs.plone.org/manage/troubleshooting/buildout.html#error-picked-some-package-some-version

```
This means that your buildout has “allow picked versions” set to false. You need to pin the version for the picked version (or turn on “allow picked versions”).
```

This was already reported 2012, but closed 2016 cf #35

### PR DESCRIPTION
**Pull request by gotcha**
It can be improved.

Fixes #481.

### CODE REVIEW

File: dev.py [hunk 1]
```diff
@@ -53,11 +53,11 @@ def install_pip():
     try:
         get_pip = os.path.join(tmp, 'get-pip.py')
         if sys.version_info < (3, ):
-            GET_PIP_URL = 'https://bootstrap.pypa.io/2.7/get-pip.py'
+            GET_PIP_URL = 'https://bootstrap.pypa.io/pip/2.7/get-pip.py'
```
Comment (jugmac00) [inline dev.py:56]: oh my! I just fixed the URL two weeks ago :-)

File: src/zc/buildout/easy_install.py [hunk 2]
```diff
@@ -1946,3 +1947,17 @@ def sort_working_set(ws, eggs_dir, develop_eggs_dir):
...
+
+The `{name}` egg does not have a version pin and allow-picked-versions = false.
+
+To resolve this, add
+
+    {name} = {version}
+
+to the [versions] section,
+
+OR set allow-picked-versions = true."""
```
Comment (jugmac00) [inline src/zc/buildout/easy_install.py:1963]: What about surrounding `allow-picked-versions = true` with e.g. single quotes?

File: src/zc/buildout/easy_install.py [hunk 3]
```diff
@@ -1946,3 +1947,17 @@ def sort_working_set(ws, eggs_dir, develop_eggs_dir):
     sorted_paths.extend(egg_paths)
     sorted_paths.extend(other_paths)
     return pkg_resources.WorkingSet(sorted_paths)
+
+
+NOT_PICKED_AND_NOT_ALLOWED = """\
+Picked: {name} = {version}
+
+The `{name}` egg does not have a version pin and allow-picked-versions = false.
```
Comment (jugmac00) [inline src/zc/buildout/easy_install.py:1955]: What about surrounding `allow-picked-versions = false` with e.g. single quotes?

File: src/zc/buildout/easy_install.py (hunk 2 above)
Comment (jugmac00) [inline src/zc/buildout/easy_install.py:1963]: I would put the the sentence after `OR` not on a single line, but just after the comma ... but that is absolutely a matter of personal style :-)

File: src/zc/buildout/tests/easy_install.txt [hunk 4]
```diff
@@ -383,6 +383,16 @@ We can request that we get an error if versions are picked:
...
+    <BLANKLINE>
+    The `demo` egg does not have a version pin and allow-picked-versions = false.
+    <BLANKLINE>
+    To resolve this, add
+    <BLANKLINE>
+        demo = 0.3
+    <BLANKLINE>
+    to the [versions] section,
+    <BLANKLINE>
+    OR set allow-picked-versions = true.
```
Comment (jugmac00) [inline src/zc/buildout/tests/easy_install.txt:395]: I can't help reviewing these doctests... but as CI is passing! Gogo!

**Review by jugmac00 — APPROVED** [review]
Thanks so much!

I just checked out your branch and ran it on locally!

```
❯ buildout
Creating directory '/tmp/buildout-test/eggs'.
Creating directory '/tmp/buildout-test/bin'.
Creating directory '/tmp/buildout-test/parts'.
Creating directory '/tmp/buildout-test/develop-eggs'.
Getting distribution for 'zc.recipe.egg>=2.0.6'.
Got zc.recipe.egg 2.0.7.
While:
  Installing.
  Getting section message.
  Initializing section message.
  Installing recipe zc.recipe.egg.
  Getting distribution for 'zc.recipe.egg>=2.0.6'.
Error: Picked: zc.recipe.egg = 2.0.7

The `zc.recipe.egg` egg does not have a version pin and allow-picked-versions = false.

To resolve this, add

    zc.recipe.egg = 2.0.7

to the [versions] section,

OR set allow-picked-versions = true.
```

This new error message is top notch!

Also thanks to @mgedmin who proposed the text.

**Comment by gotcha** [comment]
Merged locally

**Pull request closed without merging**